app/index.html

```diff
@@ -150,6 +150,58 @@
     </div>
   </ui-minimize-panel>
 
+  <ui-minimize-panel title="File list" open='false'>
+    <div class='panel panel-default'>
+      <div class='panel-heading'>
+        File List
+      </div>
+      <div class='panel-body' ng-controller="FileListController">
+        <div ng-if="file_list.files.length > 0">
+          <input type="checkbox"
+                 name="checked_files[]"
+                 value="Select all"
+                 ng-checked="all_selected"
+                 ng-click="select_all()">
+          Select all
+          <table>
+          <tr>
+            <th>Filename</th>
+            <th>Size</th>
+            <th>Last modified</th>
+            <th>Tags</th>
+          </tr>
+          <tr ng-repeat="file in file_list.files">
+            <td>
+              <input type="checkbox"
+                   name="checked_files[]"
+                   ng-checked="selected.indexOf(file.id) > -1"
+                   ng-click="toggle_file(file.id)">
+              <a href="#{{ $routeSegment.getSegmentUrl('preview', {id: file.id}) }}">{{ file.title }}</a>
+              <i class="fa fa-minus-square" style="float: right;" ng-click="remove_file(file.id)"></i>
+            </td>
+            <td>{{ file.size }}</td>
+            <td>{{ file.last_modified }}</td>
+            <td>
+              <span class="tag" ng-repeat="tag in file.tags">{{ tag }} <span ng-click="remove_tag(file.id, tag)"><i class="fa fa-minus-square"></i></span></span>
+            </td>
+          </tr>
+          </table>
+
+          <form ng-submit="submit(selected)">
+            <input type="text"
+                   ng-model="tag"
+                   ng-disabled="selected.length < 1">
+            <input type="submit"
+                   id="submit-tag"
+                   value="Add tag to selected files"
+                   ng-disabled="selected.length < 1">
+          </form>
+        </div>
+        <h3 ng-if="file_list.files.length === 0">File list is empty.</h3>
```

Comment:
Also nitpick, but `p` instead of `h3`? It's not really a header for anything, and other no-data-yet texts are `p`.